Lib/test/test_unicodedata.py

```diff
@@ -14,6 +14,10 @@
 encoding = 'utf-8'
 errors = 'surrogatepass'
 
+def all_chars():
+    '''Each Unicode codepoint, as a one-character string.'''
+    for codepoint in range(0x110000):
```

Comment:
Hum, I only expected a constant, not a whole function. I'm not sure that such function is needed, since it's only called once. I suggest to remove the function, and use sys.maxunicode rather than 0x110000. So the test will be updated automatically if sys.maxunicode is increased.

Reply:
I made it a generator because I think that makes the abstraction a bit crisper: the test just says `for char in all_chars()`, and the meaning is it iterates through all possible characters. But open-coding the loop works too; I can change it.

> So the test will be updated automatically if sys.maxunicode is increased.

I doubt very much that sys.maxunicode will ever be changed in the future. :wink: Expanding Unicode beyond the original 16 bits was traumatic enough -- it's still a widespread cause of bugs, decades later -- and when they did, they gave themselves quite a lot of safety margin.

If you'd find the code clearer to read that way, though, that'd certainly be a good reason.

FWIW I find the hex constant somewhat clearer. One reason is that I don't have to wonder about the fencepost -- note that the correct version would be `range(sys.maxunicode + 1)`.

As one form of data, a quick grep finds the existing usage in the codebase leans toward the explicit hex:
```
$ git grep maxunicode | wc -l
31

$ git grep -ie 0x10ffff -e 0x110000 | wc -l
100
```

Reply:
> I doubt very much that sys.maxunicode will ever be changed in the future.

It's not just about being future-proof: a named constant also helps to better self-explain the code, it helps the readability. I always wanted to replace hardcoded constants with named constants: I just did it in PR #15067.

Reply:
Sounds good. In this case, as I said I find the explicit literal somewhat clearer; but I definitely think that's the important criterion here, and I don't mind changing it.